packages/contracts-bedrock/scripts/L2Genesis.s.sol

```diff
@@ -0,0 +1,348 @@
+        address impl = _predeployToCodeNamespace(Predeploys.DEPLOYER_WHITELIST);
+        vm.etch(impl, vm.getDeployedCode("DeployerWhitelist.sol:DeployerWhitelist"));
+    }
+
+    /// @dev Contract metadata appended to deployed bytecode will differ
+    ///      from previous L2 genesis output
+    function _setWETH9() internal {
+        vm.etch(
+            Predeploys.WETH9,
+            vm.getDeployedCode("WETH9.sol:WETH9")
+        );
+        vm.store(
+            Predeploys.WETH9,
+            // string public name
+            hex"0000000000000000000000000000000000000000000000000000000000000000",
+            // "Wrapped Ether"
+            hex"577261707065642045746865720000000000000000000000000000000000001a"
+        );
+        vm.store(
+            Predeploys.WETH9,
+            // string public symbol
+            hex"0000000000000000000000000000000000000000000000000000000000000001",
+            // "WETH"
+            hex"5745544800000000000000000000000000000000000000000000000000000008"
+        );
+        vm.store(
+            Predeploys.WETH9,
+            // uint8 public decimals
+            hex"0000000000000000000000000000000000000000000000000000000000000002",
+            // 18
+            hex"0000000000000000000000000000000000000000000000000000000000000012"
+        );
+    }
+
+    // TODO Not sure how to set `StandardBridge public otherBridge;`
+    // L2StandardBridge's constructor doesn't take args, but calls
+    // `initialize({ _otherBridge: StandardBridge(payable(address(0))) });`
+    // which permanetly sets `StandardBridge public otherBridge`
+    function _setL2StandardBridge() internal {
+        L2StandardBridge bridge = new L2StandardBridge();
+        address impl = _predeployToCodeNamespace(Predeploys.L2_STANDARD_BRIDGE);
+
+        vm.etch(impl, address(bridge).code);
+        vm.store(
+            Predeploys.L2_STANDARD_BRIDGE,
+            // uint8 private _initialized;
+            // bool private _initializing;
+            hex"0000000000000000000000000000000000000000000000000000000000000000",
+            // 1
+            // false
+            hex"0000000000000000000000000000000000000000000000000000000000000001"
+        );
+        vm.store(
+            Predeploys.L2_STANDARD_BRIDGE,
+            // CrossDomainMessenger public messenger;
+            hex"0000000000000000000000000000000000000000000000000000000000000003",
+            // Predeploys.L2_CROSS_DOMAIN_MESSENGER
+            hex"0000000000000000000000004200000000000000000000000000000000000007"
+        );
+        vm.store(
+            Predeploys.L2_STANDARD_BRIDGE,
+            // StandardBridge public otherBridge;
+            hex"0000000000000000000000000000000000000000000000000000000000000004",
+            // L1StandardBridgeProxy
+            hex"0000000000000000000000000c8b5822b6e02cda722174f19a1439a7495a3fa6"
+        );
+
+        vm.etch(address(bridge), hex"");
+        vm.resetNonce(address(bridge));
+    }
```

Comment:
@tynes I had to manually set the storage slot for `StandardBridge public otherBridge;` because I'm unsure how to do it when deploying a new instance of `L2StandardBridge`